### DISCUSSION TITLE
F7/F8/F10/F11 keys

### DISCUSSION THREAD

**Discussion by najohnsn** [IBM/tnz#72] [Q&A] [answered]
Re: https://github.com/IBM/tnz/commit/d263f077c306a100e9b707b0179a57c2d47d6435#commitcomment-119999326
@v1gnesh

#53 is open to document this kind of thing better.

For the specific function keys in question, here is the mapping from `_termlib.py`:
```
        "\x1b[18~": "KEY_F(7)",
        "\x1b[19~": "KEY_F(8)",
        "\x1b[21~": "KEY_F(10)",
        "\x1b[23~": "KEY_F(11)",
```

So, you should be okay as long as your terminal you're running zti in generates those escape codes for your function keys. If you press one of those keys and nothing happens, there should be an entry in `tnz.log` with what code was generated.

Does that answer your question?:

**Comment by v1gnesh** [comment]
Thank you! 
I'll give it a go with https://github.com/microsoft/terminal/pull/7249 for reference.

**Comment by najohnsn** [comment, marked as answer]
Oh, Windows Terminal? I use that all the time and have no problems with F7, F8, or F10. F11 is, by default, trapped by Windows Terminal and used as a "fullscreen" key. So, F11 may require you turn that setting off. Once I went to Settings -> Actions and removed the key binding, F11 worked fine. Maybe you have key bindings set for the other function keys? In summary, I believe that Windows Terminal produces the proper escape codes... so, something else may be getting in the way.

**Reply by v1gnesh** [reply]
Ah that's right, it's actually just F11. Would be fun to setup mouse wheel down/up to F8/F7 too.
Awesome work with tnz 🤘
